Added a regex validation for admin password

MarekB01 opened the pull request:
https://github.com/BroadleafCommerce/QA/issues/4943

Created AdminUserRegexValidator to set Regex and validate password with it. Also set new properties in common.properties file, so customers could set their own regex and use this logic.

MarekB01 committed:
BroadleafCommerce/QA-4943_admin_password_regex_validation

Created AdminUserRegexValidator to set Regex and validate password with it. Also set new properties in common.properties file, so customers could set their own regex and use this logic.

oleksiimiroshnyk commented on admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/service/persistence/validation/AdminUserRegexValidator.java:
```diff
@@ -0,0 +1,75 @@
+                                             BasicFieldMetadata propertyMetadata,
+                                             String propertyName,
+                                             String value) {
+        String expression = validationConfiguration.get("regularExpression");
```
I think we don't need to get regexp from configuration validationConfiguration.get("regularExpression"); as we expect to come it from properties file
```suggestion

```

oleksiimiroshnyk commented on admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/service/persistence/validation/AdminUserRegexValidator.java:
```diff
@@ -0,0 +1,75 @@
+                                             String value) {
+        String expression = validationConfiguration.get("regularExpression");
+
+        if (value == null) {
```
I think no need to check value, which is password value for null, as we validate through regex. Or if regex validation will fail with NPE just put empty string there

MarekB01 committed:
BroadleafCommerce/QA-4943_admin_password_regex_validation

Created AdminUserRegexValidator to set Regex and validate password with it. Also set new properties in common.properties file, so customers could set their own regex and use this logic.

Commit:
BroadleafCommerce/QA#4943
- Validation added

Commit:
BroadleafCommerce/QA#4943
- Validation added

Commit:
BroadleafCommerce/QA#4943
- Validation added
